ams/AssetAnalyzer.cs

```diff
@@ -30,16 +31,40 @@ internal class AssetAnalyzer : BaseMigrator
             _tracker = tracker;
         }
 
-        private async Task<AnalysisResult> AnalyzeAsync(MediaAssetResource asset, BlobServiceClient storage, CancellationToken cancellationToken)
+        private async Task<AnalysisResult> AnalyzeAsync<T>(T item, BlobServiceClient storage, CancellationToken cancellationToken)//MediaAssetResource,BlobContainerItem
         {
-            var result = new AnalysisResult(asset.Data.Name, MigrationStatus.NotMigrated, 0);
-            _logger.LogDebug("Analyzing asset: {asset}, container: {container}", asset.Data.Name, asset.Data.Container);
+            
+            string? itemName=null;
+            string? containerName=null;
+            BlobContainerClient? container = null;
+            MediaAssetStorageEncryptionFormat? format=null;
+            if (item is MediaAssetResource mediaAsset)
+            {
+                itemName = mediaAsset.Data.Name;
+                containerName = mediaAsset.Data.Container;
+                format = mediaAsset.Data.StorageEncryptionFormat;
+                container = storage.GetContainer(mediaAsset);
+            }
+            else if (item is BlobContainerItem bcItem)
+            {
+                itemName = storage.AccountName;
```

Comment:
For AMS account, the itemName is an Asset name, but for Storage Account, the itemName is the Storage account name, this seems not match.  you may just choose container name as the item name,  to match with the asset name in AMS account.